specs/governance/alligator-interop.md

```diff
@@ -0,0 +1,136 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+  - [Interface](#interface)
+    - [Core Functions](#core-functions)
+  - [Diagram](#diagram)
+  - [Implementation](#implementation)
+  - [Backwards Compatibility](#backwards-compatibility)
+  - [Security Considerations](#security-considerations)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Overview
+
+The goal of this specification is to allow interoperability between the base and superchains for that use the [`Alligator`](alligator.md) predeploy. This allows the support of the
+subdelegation flow with for the [`GovernanceToken`](gov-token.md) by making a few changes to the existing interface for cross-chain message passing as it pertains to the checkpoint state.
+
+## Interface
+
+### Core Functions
+
+```solidity
+function afterTokenTransfer(
+    address src,
+    address dst,
+    uint256 amount
+) external
+```
+
+In addition to performing it's standard functionality this method will now use the `L2ToL2CrossDomainMessenger` to send a message to a new method on the `Alligator` contract, `afterTokenTransferInterop`. It will ensure prior to doing so that the `chainId` is not OP Mainnet i.e. 10(0xa). Moreover, it will still handle a check that both address have been migrated using
+the new storage field and update the voting power of the delegate.
+
+```solidity
+function afterTokenTransferInterop(
```

Comment:
Can we just add this to `afterTokenTransfer` instead of having 2 functions? ie update the post hook?

Reply:
Done, updates now only happens if the contract is deployed on OP mainnet